binfmt/binfmt_globals.c

```diff
@@ -56,7 +56,7 @@
  * protection to simply disable pre-emption when accessing this list.
  */
 
-FAR struct binfmt_s *g_binfmts;
+FAR struct binfmt_s *g_binfmts = NULL;
```

Comment:
should we use -fno-common option instead? There are many place which don't initialize global variables explicitly.

Reply:
i think it works. do you prefer it?

Reply:
There are many places which don't set the global variable to NULL, the same error will report again and again if we turn more option in defconfig. The right fix is to add -fno-common.

Reply:
ok, let me try the -fno-common approach later.